Bug: Type mismatch in geometries

### Describe the bug

The type of `line` in GeometryLine is defined as follows:

```ts
export class GeometryLine extends Geometry {
	readonly line: [GeometryPoint, GeometryPoint];
}
```

GeometryLine is managed as a LineString type, and according to the geo crate, it is defined as "a series of continuous line segments represented by two or more coordinates." ([ref](https://docs.rs/geo/latest/geo/#types)). However, geo ignores this definition unless necessary, so in reality, it can have fewer than two coordinates. Therefore, I think the definition should be as follows:

```ts
export class GeometryLine extends Geometry {
	readonly line: GeometryPoint[];
}
```

<details>
<summary>Running in Rust</summary>

![Screenshot from 2024-05-15 11-22-27](https://github.com/surrealdb/surrealdb.js/assets/87357925/85c428a1-bde3-436d-8485-14a8f2f33a29)
</details>

Upon examining other geometries, classes requiring similar modifications seem to be `GeometryMultiPoint`, `GeometryMultiLine`, and `GeometryMultiPolygon`.

It was expected that `Point` and `Polygon` would require one or more coordinates, but I'm unsure why `GeometryCollection` geometries would also need one or more.

<details>
<summary>Running in Rust</summary>

![Screenshot from 2024-05-15 11-40-03](https://github.com/surrealdb/surrealdb.js/assets/87357925/f78bebd1-ca4b-45b3-bede-e7173e7fcb87)
</details>

P.S. Since `GeometryPolygon` requires at least one `GeometryLine`, I think you also need to modify it to `readonly polygon: [GeometryLine, ...GeometryLine[]];`.

### Steps to reproduce

Each ring of the polygon is a LineString, and they must be closed and simple LineStrings. When an incomplete LineString is given to the polygon, geo tries to close it. Therefore, it fails the following test:

```ts
Deno.test("a", async () => {
	const surreal = await createSurreal();
	const input = new GeometryPolygon([
		new GeometryLine([
			new GeometryPoint([1, 2]),
			new GeometryPoint([3, 4]),
		]),
		new GeometryLine([
			new GeometryPoint([5, 6]),
			new GeometryPoint([7, 8]),
		]),
	]);
	const [output] = await surreal.query<[typeof input]>(
		/* surql */ `$input`,
		{ input },
	);

	assertEquals(JSON.parse(JSON.stringify(output)), {
		type: "Polygon",
		coordinates: [
			[["1", "2"], ["3", "4"], ["1", "2"]],
			[["5", "6"], ["7", "8"], ["5", "6"]],
		],
	});

	await surreal.close();
});
```

<details open>
<summary>Running in Rust</summary>

![Screenshot from 2024-05-15 12-33-02](https://github.com/surrealdb/surrealdb.js/assets/87357925/5a54805f-72e2-4b63-ac5e-a84c4447cdc8)
</details>

Test result:

```diff
a => ./tests/integration/tests/querying.ts:303:6
error: AssertionError: Values are not equal.


    [Diff] Actual / Expected


    {
      coordinates: [
        [
          [
            "1",
            "2",
          ],
          [
            "3",
            "4",
          ],
+         [
+           "1",
+           "2",
+         ],
        ],
        [
          [
            "5",
            "6",
          ],
          [
            "7",
            "8",
+         ],
+         [
+           "5",
+           "6",
          ],
        ],
      ],
      type: "Polygon",
    }
```

### Expected behaviour

It is expected that the type of the geometry class matches the actual value.

### SurrealDB version

1.5.0 for linux on x86_64

### SurrealDB.js version

v1.0.0-beta.6

### Contact Details

_No response_

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/src/library/cbor/geometry.ts
+++ b/src/library/cbor/geometry.ts
@@ -34,9 +34,11 @@ export class GeometryPoint extends Geometry {
 }
 
 export class GeometryLine extends Geometry {
-	readonly line: [GeometryPoint, GeometryPoint];
+	readonly line: [GeometryPoint, GeometryPoint, ...GeometryPoint[]];
 
-	constructor(line: [GeometryPoint, GeometryPoint] | GeometryLine) {
+	// SurrealDB only has the context of a "Line", which is two points.
+	// SurrealDB's "Line" is actually a "LineString" under the hood, which accepts two or more points
+	constructor(line: [GeometryPoint, GeometryPoint, ...GeometryPoint[]] | GeometryLine) {
 		super();
 		line = line instanceof GeometryLine ? line.line : line;
 		this.line = [new GeometryPoint(line[0]), new GeometryPoint(line[1])];

--- a/src/library/cbor/geometry.ts
+++ b/src/library/cbor/geometry.ts
@@ -38,7 +38,9 @@ export class GeometryLine extends Geometry {
 
 	// SurrealDB only has the context of a "Line", which is two points.
 	// SurrealDB's "Line" is actually a "LineString" under the hood, which accepts two or more points
-	constructor(line: [GeometryPoint, GeometryPoint, ...GeometryPoint[]] | GeometryLine) {
+	constructor(
+		line: [GeometryPoint, GeometryPoint, ...GeometryPoint[]] | GeometryLine,
+	) {
 		super();
 		line = line instanceof GeometryLine ? line.line : line;
 		this.line = [new GeometryPoint(line[0]), new GeometryPoint(line[1])];

